src/components/ShareTokenForm.js

```diff
@@ -0,0 +1,50 @@
+import { useState } from 'react';
+import { collection, getDocs, query, where } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+import useToken from '../hooks/useToken';
+
+const ShareTokenForm = () => {
+  const [error, setError] = useState(null);
+  const [selectedToken, setSelectedToken] = useState('');
+  const { setToken } = useToken();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    checkIfTokenExists();
+  };
+
+  const checkIfTokenExists = async () => {
+    const docRef = collection(db, 'List1');
+    const queryParam = query(docRef, where('token', '==', selectedToken));
+    const listSnap = await getDocs(queryParam);
+
+    if (listSnap.docs.length === 0) {
+      setError('Token does not exist please try again or create a new list');
+      setSelectedToken('');
+    } else {
+      setError(null);
+      localStorage.setItem('token', selectedToken);
+      setToken(selectedToken);
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className="form">
+      <label htmlFor="shareToken">
```

Comment:
It´s great that you took accesability into account!

Reply:
What a wonderfull job you did @redapy and @hellodeborahuk! Not only it works very well, but also you have produced a very clean, easy to follow and efficient code. Congratulation!!!